docs.go

```diff
@@ -146,3 +147,35 @@ func flagDetails(flag DocGenerationFlag) string {
 	}
 	return ": " + description
 }
+
+func prepareUsageText(command *Command) string {
```

Comment:
There's a fair bit of conditional logic in this PR for things like whether `usage` exists, whether `usage text` exists, and whether text spans multiple lines, but our current test suite doesn't capture that nuance.

Would you mind adding tests to cover the new logic?

Reply:
Yep. Updated with new tests.